rclcpp_action/src/server.cpp

```diff
@@ -343,6 +343,14 @@ ServerBase::execute_goal_request_received(std::shared_ptr<void> & data)
       response_pair.second.get());
   }
 
+  if (ret == RCL_RET_TIMEOUT) {
+    RCLCPP_WARN(
+      pimpl_->logger_,
+      "Failed to send goal response %s (timeout): %s",
+      to_string(uuid).c_str(), rcl_get_error_string().str);
+    rcl_reset_error();
+    return;
```

Comment:
I think returning here is the right thing to do. we do not need to process further since there is no client available.

Reply:
When returning RCL_RET_OK (Most cases return this), call the check function twice.  
How about combining our checks.
```
if (RCL_RET_OK != ret) {
   if (ret == RCL_RET_TIMEOUT) {
       RCLCPP_WARN(
      pimpl_->logger_,
      "Failed to send goal response %s (timeout): %s",
      to_string(uuid).c_str(), rcl_get_error_string().str);
    rcl_reset_error();
    return;
   } else {
      rclcpp::exceptions::throw_from_rcl_error(ret);
   }
}
```